src/main/java/alcazar/Parser.java

```diff
@@ -0,0 +1,192 @@
+            tasks.printEquals(search);
+            ui.showLine();
+        } else if (prompt.contains("mark") || prompt.contains("unmark") ||
+                prompt.contains("delete")) {
+            int index = Integer.parseInt(prompt.charAt(prompt.length() - 1) + "");
+            if(index > tasks.size()) {
+                throw new InvalidSerialException("☹ OOPS!!! I think you have added" +
+                        "an incorrect serial number greater than " + (tasks.size() - 1));
+            }
+            if (prompt.contains("unmark")) {
+
+                tasks.unmarkElement(index - 1);
+                storage.writeUp(tasks);
+                ui.showLine();
+                System.out.println(
+                        "OK, I've marked this task as not done yet:\n" +
+                                tasks.elementAt(index - 1) + "\n");
+                ui.showLine();
+            } else if (prompt.contains("delete")) {
+                ui.showLine();
+                System.out.print(
+                        "Noted. I've removed this task:\n" + "  " +
+                                tasks.elementAt(index - 1) + "\n");
+                tasks.delete(index - 1);
+                storage.writeUp(tasks);
+                System.out.println("Now you have " + tasks.size() + " tasks in the list\n" );
+                ui.showLine();
+            } else {
+                tasks.markElement(index - 1);
+                storage.writeUp(tasks);
+                ui.showLine();
+                System.out.println(
+                        "Nice! I've marked this task as done:\n" +
+                                tasks.elementAt(index - 1) + "\n");
+                ui.showLine();
+            }
+        } else {
+            if (prompt.contains("deadline")) {
+                String[] deadLine = this.extractDeadline(this.textAfter(prompt));
+                System.out.println(deadLine[0] + " ,, " + deadLine[1] );
+                tasks.add(new Deadline(deadLine[0], deadLine[1]));
+                storage.writeUp(tasks);
+            } else if (prompt.contains("event")) {
+                String[] eventData = this.extractEvent(this.textAfter(prompt));
+                tasks.add(new Event(eventData[2], eventData[0], eventData[1]));
+                storage.writeUp(tasks);
+            } else if (prompt.contains("todo")) {
+                tasks.add(new ToDo(this.textAfter(prompt)));
+                storage.writeUp(tasks);
+            } else {
+                throw new InvalidTaskException(
+                        "☹ OOPS!!! I'm sorry, but I don't know what that means :-("
+                );
+            }
+            ui.showLine();
+            System.out.println("Got it. I've added this task:\n "
+                    + tasks.elementAt(tasks.size() - 1) + "\n" +
+                    "Now you have " + tasks.size() + " tasks in the list\n"
+            );
+            ui.showLine();
+        }
+//        prompt = chatBot.inputText();
+    }
```

Comment:
This part of the code seems to be deep nested, making it hard to read. Would you be able to divide this part into separate methods?